### PR TITLE
feat(http-logger): support for specified the log formats via admin API

### PR DESCRIPTION
**Pull request by membphis**
### What this PR does / why we need it:

as title.

For example:
```
curl http://****/apisix/admin/plugin_metadata/http-logger -d '
{
    "log_format": {
        "host": "$host",
        "@timestamp": "$time_iso8601",
        "client_ip": "$remote_addr"
    }
}'
```

when we enabled plugin `http-logger`, we will get the message body like:

```
{"host":"localhost","@timestamp":"2020-09-23T18:29:07-04:00","client_ip":"127.0.0.1","route_id":"1"}
{"host":"localhost","@timestamp":"2020-09-23T18:29:07-04:00","client_ip":"127.0.0.1","route_id":"1"}
```

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Review by nic-chen — APPROVED** [review]

**Comment by moonming** [comment]
why not add `log_format` in http-logger plugin directly?

**Comment by moonming** [comment]
Different routes may have different log formats

File: bin/apisix [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ local function init_etcd(show_output)
         for _, dir_name in ipairs({"/routes", "/upstreams", "/services",
                                    "/plugins", "/consumers", "/node_status",
                                    "/ssl", "/global_rules", "/stream_routes",
-                                   "/proto"}) do
+                                   "/proto", "/plugin_metadata"}) do
```
Comment (moonming) [inline bin/apisix:1028]: What is the difference between metadata and the properties of plugin itself?  I didn't see why we need it.

Comment (membphis) [reply bin/apisix:1028]: get plugin schema: 
/apisix/admin/plugins/{plugin-name}, eg: /apisix/admin/plugins/limit-req

get plugin metadata:
/apisix/admin/plugin_metadata/{plugin-name}, eg: /apisix/admin/plugin_metadata/example-plugin

Comment (moonming) [reply bin/apisix:1028]: https://github.com/apache/apisix/pull/2294#issuecomment-698108183

Comment (moonming) [reply bin/apisix:1028]: why we need this PR?

Comment (juzhiyuan) [reply bin/apisix:1028]: I would recommend using `/plugins/{name}/metadat` in a RESTful way if this PR is needed..

Comment (ShiningRush) [reply bin/apisix:1028]: I am also curious about what scenarios need metadata instead of plugin's option.

Comment (membphis) [reply bin/apisix:1028]: this PR has been merged. we can talk in here: https://github.com/apache/apisix/pull/2307#issuecomment-698337991

**Pull request merged**